File: cross_platform_build/lib/views/element_selection_view.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:provider/provider.dart';
import '../managers/user_profile_manager.dart';
import '../models/lote_models.dart';

class ElementSelectionView extends StatefulWidget {
  const ElementSelectionView({super.key});

  @override
  State<ElementSelectionView> createState() => _ElementSelectionViewState();
}

class _ElementSelectionViewState extends State<ElementSelectionView> {
  late int _localElementIndex;
  late ExpressionStyle _localExpression;

  @override
  void initState() {
    super.initState();
    final profile = Provider.of<UserProfileManager>(context, listen: false);
    _localElementIndex = profile.selectedElementIndex;
    _localExpression = profile.expressionStyle;
  }

  LotEElement get _activeElement => UserProfileManager.availableElements[_localElementIndex];

  void _saveSelection(UserProfileManager profile) {
    profile.selectedElementIndex = _localElementIndex;
    profile.expressionStyle = _localExpression;
    Navigator.of(context).pop();
  }

  @override
  Widget build(BuildContext context) {
    final profile = Provider.of<UserProfileManager>(context);

    return Scaffold(
      backgroundColor: const Color(0xFF050505),
      appBar: AppBar(
        backgroundColor: const Color(0xFF050505),
        elevation: 0,
        leading: IconButton(
          icon: const Icon(Icons.close, color: Colors.white),
          onPressed: () => Navigator.of(context).pop(),
        ),
      ),
      body: Stack(
        children: [
          // Background ambient glow
          Positioned(
            left: -50,
            top: 20,
            child: Container(
              width: 320,
              height: 320,
              decoration: BoxDecoration(
                shape: BoxShape.circle,
                color: _activeElement.primaryColor.withOpacity(0.08),
              ),
            ),
          ),
          SafeArea(
            child: Column(
              children: [
                // Header
                Column(
                  children: [
                    Text(
                      "ELEMENT THEME SELECTION",
                      style: GoogleFonts.orbitron(
                        fontSize: 20,
                        fontWeight: FontWeight.bold,
                        letterSpacing: 3,
                        color: Colors.white,
                      ),
                    ),
                    const SizedBox(height: 6),
                    Text(
                      "Choose your elemental theme and stance",
                      style: GoogleFonts.exo2(
                        fontSize: 13,
                        color: Colors.grey,
                      ),
                    ),
                  ],
                ),
                const SizedBox(height: 20),

                // Horizontal Elements Scroll
                SizedBox(
                  height: 50,
                  child: ListView.builder(
                    scrollDirection: Axis.horizontal,
                    padding: const EdgeInsets.symmetric(horizontal: 16),
                    itemCount: UserProfileManager.availableElements.length,
                    itemBuilder: (context, idx) {
                      final elem = UserProfileManager.availableElements[idx];
                      final isSelected = _localElementIndex == idx;

                      return Padding(
                        padding: const EdgeInsets.only(right: 10),
                        child: InkWell(
                          onTap: () {
                            setState(() {
                              _localElementIndex = idx;
                              if (elem.inherentDark) {
                                _localExpression = ExpressionStyle.corrupt;
                              }
                            });
                          },
                          borderRadius: BorderRadius.circular(10),
                          child: Container(
                            alignment: Alignment.center,
                            padding: const EdgeInsets.symmetric(horizontal: 16),
                            decoration: BoxDecoration(
                              color: isSelected ? elem.primaryColor : Colors.white.withOpacity(0.03),
                              borderRadius: BorderRadius.circular(10),
                              border: Border.all(
                                color: isSelected ? elem.accentColor : Colors.white.withOpacity(0.1),
                                width: 1,
                              ),
                              boxShadow: isSelected
                                  ? [
                                      BoxShadow(
                                        color: elem.primaryColor.withOpacity(0.4),
                                        blurRadius: 6,
                                      ),
                                    ]
                                  : null,
                            ),
                            child: Text(
                              elem.name.toUpperCase(),
                              style: GoogleFonts.orbitron(
                                fontSize: 12,
                                fontWeight: FontWeight.bold,
                                color: isSelected ? Colors.white : Colors.grey,
                              ),
                            ),
                          ),
                        ),
                      );
                    },
                  ),
                ),
                const SizedBox(height: 20),

                // Content scroll
                Expanded(
                  child: SingleChildScrollView(
                    padding: const EdgeInsets.symmetric(horizontal: 16),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        // Expression stance select buttons
                        Text(
                          "ELEMENTAL EXPRESSION STANCE",
                          style: GoogleFonts.orbitron(
                            fontSize: 11,
                            fontWeight: FontWeight.bold,
                            color: Colors.grey,
                            letterSpacing: 2,
                          ),
                        ),
                        const SizedBox(height: 10),
                        Row(
                          children: ExpressionStyle.values.map((style) {
                            final isSel = _localExpression == style;
                            final isDisabled = _activeElement.inherentDark && style != ExpressionStyle.corrupt;

                            return Expanded(
                              child: Padding(
                                padding: const EdgeInsets.symmetric(horizontal: 4),
                                child: InkWell(
                                  onTap: isDisabled
                                      ? null
                                      : () {
                                          setState(() {
                                            _localExpression = style;
                                          });
                                        },
                                  borderRadius: BorderRadius.circular(8),
                                  child: Container(
                                    alignment: Alignment.center,
                                    padding: const EdgeInsets.symmetric(vertical: 12),
                                    decoration: BoxDecoration(
                                      color: isSel
                                          ? _activeElement.primaryColor
                                          : Colors.white.withOpacity(0.02),
                                      borderRadius: BorderRadius.circular(8),
                                      border: Border.all(
                                        color: isSel
                                            ? _activeElement.accentColor
                                            : Colors.white.withOpacity(0.08),
                                        width: 1,
                                      ),
                                    ),
                                    child: Text(
                                      style.displayName,
                                      style: GoogleFonts.exo2(
                                        fontSize: 10,
                                        fontWeight: FontWeight.bold,
                                        color: isSel
                                            ? Colors.white
                                            : (isDisabled ? Colors.white.withOpacity(0.15) : Colors.grey),
                                      ),
                                    ),
                                  ),
                                ),
                              ),
                            );
                          }).toList(),
                        ),
                        if (_activeElement.inherentDark) ...[
                          const SizedBox(height: 8),
                          Text(
                            "⚠️ Tenebrie elements are inherently corruptive. Standard Light alignment is locked by lore.",
                            style: GoogleFonts.exo2(
                              fontSize: 10,
                              color: Colors.red.withOpacity(0.8),
                            ),
                          ),
                        ],
                        const SizedBox(height: 25),

                        // Element Lore Card
                        Container(
                          padding: const EdgeInsets.all(20),
                          decoration: BoxDecoration(
                            color: Colors.white.withOpacity(0.02),
                            borderRadius: BorderRadius.circular(16),
                            border: Border.all(
                              color: _activeElement.primaryColor.withOpacity(0.3),
                              width: 1.5,
                            ),
                          ),
                          child: Column(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: [
                              Row(
                                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                                children: [
                                  Text(
                                    _activeElement.displayName(_localExpression).toUpperCase(),
                                    style: GoogleFonts.orbitron(
                                      fontSize: 18,
                                      fontWeight: FontWeight.bold,
                                      color: _activeElement.primaryColor,
                                      letterSpacing: 2,
                                    ),
                                  ),
                                  Container(
                                    padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                                    decoration: BoxDecoration(
                                      color: Colors.white.withOpacity(0.05),
                                      borderRadius: BorderRadius.circular(6),
                                    ),
                                    child: Text(
                                      _activeElement.planetOfOrigin.toUpperCase(),
                                      style: GoogleFonts.orbitron(
                                        fontSize: 9,
                                        fontWeight: FontWeight.bold,
                                        color: Colors.grey,
                                      ),
                                    ),
                                  )
                                ],
                              ),
                              const SizedBox(height: 15),
                              Text(
                                _activeElement.description,
                                style: GoogleFonts.exo2(
                                  fontSize: 13.5,
                                  color: Colors.white,
                                  height: 1.5,
                                ),
                              ),
                              const SizedBox(height: 15),
                              const Divider(color: Colors.white10),
                              const SizedBox(height: 10),
                              Text(
                                "STANCE MANIFESTATION",
                                style: GoogleFonts.orbitron(
                                  fontSize: 11,
                                  fontWeight: FontWeight.bold,
                                  color: Colors.grey,
                                  letterSpacing: 2,
                                ),
                              ),
                              const SizedBox(height: 8),
                              Builder(
                                builder: (context) {
                                  final stanceText = _localExpression == ExpressionStyle.standard
                                      ? _activeElement.standardDetails
                                      : _localExpression == ExpressionStyle.corrupt
                                          ? _activeElement.corruptDetails
                                          : _activeElement.balancedDetails;
                                  return Text(
                                    stanceText,
                                    style: GoogleFonts.exo2(
                                      fontSize: 13,
                                      color: Colors.white.withOpacity(0.8),
                                      height: 1.5,
                                    ),
                                  );
                                },
                              )
                            ],
                          ),
                        ),
                        const SizedBox(height: 25),

                        // Action Button
                        SizedBox(
                          width: double.infinity,
                          child: ElevatedButton(
                            onPressed: () => _saveSelection(profile),
                            style: ElevatedButton.styleFrom(
                              backgroundColor: _activeElement.primaryColor,
                              padding: const EdgeInsets.symmetric(vertical: 14),
                              shape: RoundedRectangleBorder(
                                borderRadius: BorderRadius.circular(10),
                              ),
                              elevation: 6,
                              shadowColor: _activeElement.primaryColor.withOpacity(0.4),
                            ),
                            child: Text(
                              "CONFIRM ELEMENT THEME",
                              style: GoogleFonts.orbitron(
                                fontSize: 15,
                                fontWeight: FontWeight.bold,
                                letterSpacing: 2,
                                color: Colors.white,
                              ),
                            ),
                          ),
                        ),
                        const SizedBox(height: 30),
                      ],
                    ),
                  ),
                )
              ],
            ),
          )
        ],
      ),
    );
  }
}

File: LotE Workout Companion/Views/ElementSelectionView.swift
//
//  ElementSelectionView.swift
//  LotE Workout Companion
//
//  Created by Antigravity on 6/23/26.
//

import SwiftUI

struct ElementSelectionView: View {
    @ObservedObject var profileManager: UserProfileManager
    @Environment(\.presentationMode) var presentationMode
    
    @State private var localElementIndex: Int = 0
    @State private var localExpression: ExpressionStyle = .standard
    
    var activeElement: LotEElement {
        UserProfileManager.availableElements[localElementIndex]
    }
    
    var body: some View {
        ZStack {
            // Element themed dark background
            Color(hex: "#050505")
                .ignoresSafeArea()
            
            // Dynamic thematic background glow
            Circle()
                .fill(activeElement.primaryColor.opacity(0.12))
                .frame(width: 320, height: 320)
                .blur(radius: 90)
                .position(x: 100, y: 150)
                .ignoresSafeArea()
            
            VStack(spacing: 0) {
                // Header
                VStack(spacing: 8) {
                    Text("RESONANCE CHAMBER")
                        .font(.custom("Orbitron-Bold", size: 22).bold())
                        .foregroundColor(.white)
                        .tracking(3)
                    
                    Text("Tune your elemental wavelength and stance")
                        .font(.custom("Exo2-Medium", size: 13))
                        .foregroundColor(.gray)
                }
                .padding(.top, 25)
                .padding(.bottom, 15)
                
                // Horizontal scroll list of elements
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 14) {
                        ForEach(0..<UserProfileManager.availableElements.count, id: \.self) { idx in
                            let elem = UserProfileManager.availableElements[idx]
                            let isSelected = localElementIndex == idx
                            
                            Button(action: {
                                withAnimation(.spring()) {
                                    localElementIndex = idx
                                    // Handle Tenebrie genetic enforcement
                                    if elem.inherentDark {
                                        localExpression = .corrupt
                                    } else if localExpression == .corrupt && elem.name == "Water" {
                                        localExpression = .corrupt
                                    }
                                }
                            }) {
                                Text(elem.name.uppercased())
                                    .font(.custom("Orbitron-Bold", size: 12).bold())
                                    .foregroundColor(isSelected ? .white : .gray)
                                    .padding(.vertical, 10)
                                    .padding(.horizontal, 16)
                                    .background(
                                        RoundedRectangle(cornerRadius: 10)
                                            .fill(isSelected ? elem.primaryColor : Color.white.opacity(0.03))
                                            .overlay(
                                                RoundedRectangle(cornerRadius: 10)
                                                    .stroke(isSelected ? elem.accentColor : Color.white.opacity(0.1), lineWidth: 1)
                                            )
                                    )
                                    .shadow(color: isSelected ? elem.primaryColor.opacity(0.4) : Color.clear, radius: 4)
                            }
                        }
                    }
                    .padding(.horizontal)
                }
                .frame(height: 50)
                .padding(.bottom, 15)
                
                ScrollView {
                    VStack(spacing: 20) {
                        // Expression Selector Stance (Standard vs Corrupted vs Balanced)
                        VStack(alignment: .leading, spacing: 10) {
                            Text("ELEMENTAL EXPRESSION STANCE")
                                .font(.custom("Orbitron-Bold", size: 12).bold())
                                .foregroundColor(.gray)
                                .tracking(2)
                                .padding(.horizontal)
                            
                            HStack(spacing: 10) {
                                ForEach(ExpressionStyle.allCases, id: \.self) { style in
                                    let isSel = localExpression == style
                                    let isDisabled = activeElement.inherentDark && style != .corrupt
                                    
                                    Button(action: {
                                        if !isDisabled {
                                            localExpression = style
                                        }
                                    }) {
                                        Text(style.rawValue)
                                            .font(.custom("Exo2-Bold", size: 11))
                                            .foregroundColor(isSel ? .white : (isDisabled ? .white.opacity(0.15) : .gray))
                                            .padding(.vertical, 12)
                                            .frame(maxWidth: .infinity)
                                            .background(
                                                RoundedRectangle(cornerRadius: 8)
                                                    .fill(isSel ? activeElement.primaryColor : Color.white.opacity(0.02))
                                                    .overlay(
                                                        RoundedRectangle(cornerRadius: 8)
                                                            .stroke(isSel ? activeElement.accentColor : Color.white.opacity(0.08), lineWidth: 1)
                                                    )
                                            )
                                    }
                                    .disabled(isDisabled)
                                }
                            }
                            .padding(.horizontal)
                            
                            if activeElement.inherentDark {
                                Text("⚠️ Tenebrie elements are inherently corruptive. Standard Light alignment is locked by lore.")
                                    .font(.caption2)
                                    .foregroundColor(.red.opacity(0.8))
                                    .padding(.horizontal)
                                    .padding(.top, 4)
                            }
                        }
                        
                        // Element Lore Card
                        VStack(alignment: .leading, spacing: 15) {
                            HStack {
                                Text(activeElement.displayName(for: localExpression).uppercased())
                                    .font(.custom("Orbitron-Bold", size: 20).bold())
                                    .foregroundColor(activeElement.primaryColor)
                                    .tracking(2)
                                
                                Spacer()
                                
                                Text(activeElement.planetOfOrigin.uppercased())
                                    .font(.custom("Orbitron-Bold", size: 11).bold())
                                    .foregroundColor(.gray)
                                    .padding(.vertical, 4)
                                    .padding(.horizontal, 8)
                                    .background(Color.white.opacity(0.05))
                                    .cornerRadius(6)
                            }
                            
                            // Cultural Description
                            Text(activeElement.description)
                                .font(.custom("Exo2-Medium", size: 14))
                                .foregroundColor(.white)
                                .lineSpacing(4)
                            
                            Divider()
                                .background(Color.white.opacity(0.1))
                            
                            // Stance Details
                            VStack(alignment: .leading, spacing: 8) {
                                Text("STANCE MANIFESTATION")
                                    .font(.custom("Orbitron-Bold", size: 11).bold())
                                    .foregroundColor(.gray)
                                    .tracking(2)
                                
                                let stanceText: String = {
                                    switch localExpression {
                                    case .standard: return activeElement.standardDetails
                                    case .corrupt: return activeElement.corruptDetails
                                    case .balanced: return activeElement.balancedDetails
                                    }
                                }()
                                
                                Text(stanceText)
                                    .font(.custom("Exo2-Regular", size: 13))
                                    .foregroundColor(.white.opacity(0.8))
                                    .lineSpacing(4)
                            }
                        }
                        .padding(20)
                        .background(
                            RoundedRectangle(cornerRadius: 16)
                                .fill(Color.white.opacity(0.02))
                                .overlay(
                                    RoundedRectangle(cornerRadius: 16)
                                        .stroke(activeElement.primaryColor.opacity(0.3), lineWidth: 1.5)
                                )
                        )
                        .padding(.horizontal)
                        
                        // Save Button
                        Button(action: {
                            saveSelection()
                        }) {
                            Text("ALIGN SOUL CHANNELS")
                                .font(.custom("Orbitron-Bold", size: 15).bold())
                                .foregroundColor(.white)
                                .tracking(2)
                                .padding(.vertical, 14)
                                .frame(maxWidth: .infinity)
                                .background(
                                    RoundedRectangle(cornerRadius: 10)
                                        .fill(activeElement.primaryColor)
                                        .shadow(color: activeElement.primaryColor.opacity(0.4), radius: 6)
                                )
                        }
                        .padding(.horizontal, 20)
                        .padding(.top, 10)
                        .padding(.bottom, 30)
                    }
                }
            }
        }
        .onAppear {
            localElementIndex = profileManager.selectedElementIndex
            localExpression = profileManager.expressionStyle
        }
    }
    
    private func saveSelection() {
        profileManager.selectedElementIndex = localElementIndex
        profileManager.expressionStyle = localExpression
        presentationMode.wrappedValue.dismiss()
    }
}
